content/blog/2022-05-05-end-to-end-computer-vision-api-part-2-local-experiments.md

```diff
@@ -0,0 +1,207 @@
+---
+title: 'End-to-End Computer Vision API, Part 2: Local Experiments'
+date: 2022-05-05
+description: >
+  In part 1, we talked about effective management and versioning of large
+  datasets and the creation of reproducible ML pipelines.
+
+  Here we'll learn about experiment management:  generation of many experiments
+  by tweaking configurations and hyperparameters;  comparison of experiments
+  based on their performance metrics; and persistence of the most promising ones
+
+descriptionLong: |
+  _This is the second part of a three-part series of posts._
+
+  In [part 1](https://dvc.org/blog/end-to-end-computer-vision-api-part-1-data-versioning-and-ml-pipelines) of this series of posts, we introduced a solution to a common problem faced by companies 
+  in the manufacturing industry: detecting defects from images of products moving along a production line. 
+  The solution we proposed was a Deep Learning-based image segmentation model wrapped in a web API. 
+  We talked about effective management and versioning of large datasets and the creation of reproducible ML pipelines.
+  Here we'll learn about experiment management: 
+  generation of many experiments by tweaking configurations and hyperparameters; 
+  comparison of experiments based on their performance metrics; and
+  persistence of the most promising ones
+
+picture: 2022-05-05/e2e-cv-pt2-cover.png
+author: alex_kim
+commentsUrl: https://discuss.dvc.org/t/end-to-end-computer-vision/1178
+tags:
+  - Computer Vision
+  - DVC
+  - CML
+  - Studio
+  - CI/CD
+  - Experiment Tracking
+---
+
+### Introduction
+
+[Earlier](https://dvc.org/blog/end-to-end-computer-vision-api-part-1-data-versioning-and-ml-pipelines),
+we built a pipeline that produces a trained Computer Vision model. Now we need a
+way to efficiently tune its configuration and the hyperparameters of the model.
+We want the ability to:
+
+- Run many experiments and easily compare their results to pick the
+  best-performing ones.
+- Track the global history of the model's performance, and map each improvement
+  to a particular change in code, configuration, or data.
+- Zoom into the details of each training run to help us diagnose issues.
+
+### Experiment Management
+
+Our DVC pipeline relies on the parameters defined in the
+[`params.yaml`](https://github.com/iterative/magnetic-tiles-defect/blob/main/params.yaml)
+file. By loading its contents in each stage, we can avoid hard-coded parameters.
```

Comment:
Perhaps it's my understanding but it reads to me like DVC *requires* this file (because of *relies*) and somehow implies that DVC only supports `params.yaml` for configuration, when it supports arbitrary numbers and names of parameters files (from https://dvc.org/doc/command-reference/params#description):

> The default parameters file name is params.yaml, but any other YAML 1.2, JSON, TOML, or [Python](https://dvc.org/doc/command-reference/params#examples-python-parameters-file) files can be used additionally (listed under params: with a sub-list of param values, as shown in the sample above) .

Sorry for late comment 🙏 Not critical, can be released without addressing

Reply:
Good point @daavoo!  Maybe should change to 

"Our DVC pipeline relies on the parameters defined in the[`params.yaml]( https://github.com/iterative/magnetic-tiles-defect/blob/main/params.yaml) file in this case (see [other possible file types here]( https://dvc.org/doc/command-reference/params#description)."

That way it keeps readers on track, but if they are interested in other types they know it and can get info on it.

Reply:
Good point! Updated with @jendefig's suggestion.